### PR TITLE
Remove redundant fit call from accelerator connector test

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Removes a redundant fit call from a test that does not need it for its assertions. 
We have seen this test stall on a few PRs in the CI (not reproducible locally).
With a bit of luck, the removal of this fit call will (temporarily) unblock us.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

File: tests/accelerators/test_accelerator_connector.py [hunk 1]
```diff
@@ -350,7 +350,6 @@ def test_accelerator_choice_ddp_cpu_and_strategy_spawn(tmpdir):
 
 
 def _test_accelerator_choice_ddp_cpu_and_strategy(tmpdir, ddp_strategy_class):
-    model = BoringModel()
     trainer = Trainer(
         default_root_dir=tmpdir,
         strategy=ddp_strategy_class(find_unused_parameters=True),
```
Comment (carmocca) [inline tests/accelerators/test_accelerator_connector.py:355]: I'm not sure if this is related to the intermittent failures at all, but this test gets a `strategy` AND sets `accelerator="ddp_cpu"` which signals the strategy spawn.

IMO we should change it to `accelerator="cpu"`.

(if done, remember to update the function name)

**Review by carmocca — APPROVED** [review]

File: tests/accelerators/test_accelerator_connector.py (hunk 1 above)
Comment (awaelchli) [reply tests/accelerators/test_accelerator_connector.py:355]: there are many other usages of accelerator="ddp_cpu" and we haven't deprecated that yet (afaik). So we should leave it until that day comes.

Comment (carmocca) [reply tests/accelerators/test_accelerator_connector.py:355]: I'm not saying we need to deprecate it. I mean that with the parameterizations, this becomes

(1):
```python
strategy=DDPSpawnPlugin(),
accelerator="ddp_cpu"  # equals to `strategy="ddp_spawn", accelerator="cpu"`
```
which I guess it's okay since there is no conflict


(2):
```python
strategy=DDPlugin(),
accelerator="ddp_cpu"  # equals to `strategy="ddp_spawn", accelerator="cpu"`
```

where there is a conflict, which one takes precedence, "DDPPlugin" or "ddp_spawn"?

Is the test written like this on purpose?

Comment (awaelchli) [reply tests/accelerators/test_accelerator_connector.py:355]: Yes it is. There was a time where we didn't support (2) and I had to add it: #6208
That's basically the test for this functionality.

Comment (carmocca) [reply tests/accelerators/test_accelerator_connector.py:355]: would you mind adding an assertion for the expected TTP class then?

Comment (awaelchli) [reply tests/accelerators/test_accelerator_connector.py:355]: I'm not sure what you mean. We have this assertion:

```
assert isinstance(trainer.training_type_plugin, ddp_strategy_class)
```

ddp_strategy_class is either DDPPlugin or DDPSpawnPlugin
while accelerator="ddp_cpu"

**Review by kaushikb11 — APPROVED** [review]
